samza-core/src/main/java/org/apache/samza/coordinator/stream/CoordinatorStreamManager.java

```diff
@@ -0,0 +1,197 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator.stream;
+
+import org.apache.samza.config.Config;
+import org.apache.samza.coordinator.stream.messages.CoordinatorStreamMessage;
+
+import java.util.Set;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Class which handles the common functionality for coordinator stream consumer and producer
+ */
+public class CoordinatorStreamManager {
+  private Logger log = LoggerFactory.getLogger(CoordinatorStreamManager.class);
+  private final String source;
+  private CoordinatorStreamSystemProducer coordinatorStreamProducer;
+  private CoordinatorStreamSystemConsumer coordinatorStreamConsumer;
+
+  /**
+   * Creates a new {@link CoordinatorStreamManager} with a given coordinator stream producer, consumer and with a given source.
+   *
+   * @param coordinatorStreamProducer The {@link CoordinatorStreamSystemProducer} which should be used with the {@link CoordinatorStreamManager}
+   * @param coordinatorStreamConsumer The {@link CoordinatorStreamSystemConsumer} which should be used with the {@link CoordinatorStreamManager}
+   * @param source The source of the coordinator stream.
+   */
+  public CoordinatorStreamManager(CoordinatorStreamSystemProducer coordinatorStreamProducer, CoordinatorStreamSystemConsumer coordinatorStreamConsumer, String source) {
+    this.coordinatorStreamProducer = coordinatorStreamProducer;
+    this.coordinatorStreamConsumer = coordinatorStreamConsumer;
+    this.source = source;
+  }
+
+  /**
+   * Creates a new {@link CoordinatorStreamManager} and instantiates the underlying coordinator stream producer and consumer.
+   * @param coordinatorSystemConfig Configuration used to instantiate the coordinator stream producer and consumer.
+   * @param metricsRegistry Metrics registry
+   * @param source The source of the coordinator stream.
+   */
+  public CoordinatorStreamManager(Config coordinatorSystemConfig, MetricsRegistry metricsRegistry, String source) {
+    coordinatorStreamConsumer = new CoordinatorStreamSystemConsumer(coordinatorSystemConfig, metricsRegistry);
+    coordinatorStreamProducer = new CoordinatorStreamSystemProducer(coordinatorSystemConfig, metricsRegistry);
+    this.source = source;
+  }
+
+  /**
+   * Fully starts, registers and bootstraps the underlying coordinator stream producer and consumer.
+   */
+  public void registerStartBootstrapAll() {
```

Comment:
This is a weird method. Should these be separate register, start and bootstrap methods?